api/v1alpha1/obclusteroperation_types.go

```diff
@@ -0,0 +1,119 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apitypes "github.com/oceanbase/ob-operator/api/types"
+	tasktypes "github.com/oceanbase/ob-operator/pkg/task/types"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// OBClusterOperationSpec defines the desired state of OBClusterOperation
+type OBClusterOperationSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	OBCluster          string                        `json:"obcluster"`
+	Type               apitypes.ClusterOperationType `json:"type"`
+	Force              bool                          `json:"force,omitempty"`
+	AddZones           []apitypes.OBZoneTopology     `json:"addZones,omitempty"`
+	DeleteZones        []string                      `json:"deleteZones,omitempty"`
+	AdjustReplicas     []AlterZoneReplicas           `json:"adjustReplicas,omitempty"`
+	RestartOBServers   *RestartOBServersConfig       `json:"restartOBServers,omitempty"`
+	Upgrade            *UpgradeConfig                `json:"upgrade,omitempty"`
+	ModifyStorageClass *ModifyStorageClassConfig     `json:"modifyStorageClass,omitempty"`
```

Comment:
modify storage class and expand storage class both influence observer, so they can be moved under ModifyOBServer